modules/mcuboot/CMakeLists.txt

```diff
@@ -28,6 +28,10 @@ endif()
 
 if(CONFIG_BOOTLOADER_MCUBOOT)
   if (CONFIG_NCS_IS_VARIANT_IMAGE)
+    # NCS Handles everything regarding mcuboot, ensure Zephyr doesn't interfere.
+    # This is a temporary solution until Zephyr signing has been made more modular.
+    set(CONFIG_BOOTLOADER_MCUBOOT False PARENT_SCOPE)
```

Comment:
this is not safe.

CONFIG_BOOTLOADER_MCUBOOT is a setting imported from the `.config`.
I acknowledge that it only impacts CMake configure time, and that CONFIG_BOOTLOADER_MCUBOOT is still set in autoconf.h and elsewhere in the system, such as python code loading the `.config`.

But in CMake, there can be customer code checking for this flag for other purposes than signing.

Has it been investigated why this happens for XIP images but not for other variant builds ?

Reply:
Hi @tejlmand,

Yes, I am aware that the solution is not perfect and we should not manipulate Kconfig values inside CMake. I just make the behavior consistent with handling main application image - I actually used the same workaround (please take a look at the bottom of this CMakeLists.txt file, you will see the same hotfix applied for the main image):
```
# NCS Handles everything regarding mcuboot, ensure Zephyr doesn't interfere.
# This is a temporary solution until Zephyr signing has been made more modular.
set(CONFIG_BOOTLOADER_MCUBOOT False PARENT_SCOPE)
```

Reply:
Could you provide a proper solution for the problem during the ongoing upmerge? Or maybe we could accept my temporary workaround and improve it later on (together with improving integration for the main application image)?

Reply:
I see that.
Not fond of that solution either, but at least you are consistent with what we are already doing.

Solution accepted, but please do a rebase.

Reply:
I need to open a new PR to sdk-nrf main branch. Give me moment.